### ISSUE TITLE
building master with --disable-mpi-io fails

### ISSUE THREAD

**Issue by tkordenbrock** [open-mpi/ompi#1932]
When building the latest master using --disable-mpi-io, I get these errors.

```
  CC       libmca_common_ompio_la-common_ompio_print_queue.lo
  CC       libmca_common_ompio_la-common_ompio_file_open.lo
../../../../../../ompi/mca/common/ompio/common_ompio_file_open.c: In function 'mca_common_ompio_file_open':
../../../../../../ompi/mca/common/ompio/common_ompio_file_open.c:56:18: error: 'MPI_MODE_RDONLY' undeclared (first use in this function)
     if ( ((amode&MPI_MODE_RDONLY)?1:0) + ((amode&MPI_MODE_RDWR)?1:0) +
                  ^
../../../../../../ompi/mca/common/ompio/common_ompio_file_open.c:56:18: note: each undeclared identifier is reported only once for each function it appears in
../../../../../../ompi/mca/common/ompio/common_ompio_file_open.c:56:50: error: 'MPI_MODE_RDWR' undeclared (first use in this function)
     if ( ((amode&MPI_MODE_RDONLY)?1:0) + ((amode&MPI_MODE_RDWR)?1:0) +
                                                  ^
../../../../../../ompi/mca/common/ompio/common_ompio_file_open.c:57:11: error: 'MPI_MODE_WRONLY' undeclared (first use in this function)
   ((amode&MPI_MODE_WRONLY)?1:0) != 1 ) {
           ^
../../../../../../ompi/mca/common/ompio/common_ompio_file_open.c:62:19: error: 'MPI_MODE_CREATE' undeclared (first use in this function)
         ((amode & MPI_MODE_CREATE) || (amode & MPI_MODE_EXCL))) {
                   ^
../../../../../../ompi/mca/common/ompio/common_ompio_file_open.c:62:48: error: 'MPI_MODE_EXCL' undeclared (first use in this function)
         ((amode & MPI_MODE_CREATE) || (amode & MPI_MODE_EXCL))) {
                                                ^
../../../../../../ompi/mca/common/ompio/common_ompio_file_open.c:66:45: error: 'MPI_MODE_SEQUENTIAL' undeclared (first use in this function)
     if ((amode & MPI_MODE_RDWR) && (amode & MPI_MODE_SEQUENTIAL)) {
                                             ^
../../../../../../ompi/mca/common/ompio/common_ompio_file_open.c:208:30: error: 'MPI_MODE_APPEND' undeclared (first use in this function)
     if ( ompio_fh->f_amode & MPI_MODE_APPEND ) {
                              ^
../../../../../../ompi/mca/common/ompio/common_ompio_file_open.c: In function 'mca_common_ompio_file_close':
../../../../../../ompi/mca/common/ompio/common_ompio_file_open.c:255:30: error: 'MPI_MODE_DELETE_ON_CLOSE' undeclared (first use in this function)
     if ( ompio_fh->f_amode & MPI_MODE_DELETE_ON_CLOSE ) {
                              ^

```

**Comment by edgargabriel** [comment]
I'll have a look at it, I did not try with --disable-mpi-io, but its a fair point

**Comment by edgargabriel** [comment]
@jsquyres I have a question to you: do I remember our discussion correctly, that a subdirectory in common always has to be compiled? So the question is, do we need to modify the configury script to exclude common/ompio in case the user requested --disable-mpi-io, or should I just put the source files in common/ompio/Makefile.am in a conditional statement (similarly how the MPI_File_* functions in ompi/.mpi/c/ are provided) and thus create an empty .so file?

**Comment by jsquyres** [comment]
Each directory must be *traversed*, even if the sources in that directory are not *compiled*.

So your latter suggestion is the correct one: put the sources *and* the library in that `common/ompio/Makefile.am` in an `AM_CONDITIONAL`.  That way, the library will not be built in the `--disable-mpi-io` case, but `make dist` will still pick up all the sources.

**Comment by edgargabriel** [comment]
ok, thank you. I think I have a working solution, will push iot to master later this afternoon.

**Comment by jjhursey** [comment]
Somewhat related to this, it would be useful to have a:
```shell
  --disable-io-ompio      Disable the OMPIO MPI-IO components
```
Similar to the one for ROMIO
```shell
  --disable-io-romio      Disable the ROMIO MPI-IO component
```

Particularly since OMPIO has a number of frameworks that are related to it. It would make it easier to turn off this functionality via configure instead of using:
```shell
./configure --enable-mca-no-build=io-ompio,common-ompio,sharedfp,fbtl,fcoll 
```

**Comment by edgargabriel** [comment]
@jjhursey we can open a new issue for that
